File: LC_JS-SOLUTIONS/560. Subarray Sum Equals K.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function (nums, k) {
  let count = 0;
  let prefixSum = new Map();
  let currSum = 0;
  prefixSum.set(0, 1);

  for (let i = 0; i < nums.length; i++) {
    currSum += nums[i];
    if (prefixSum.has(currSum - k)) {
      count += prefixSum.get(currSum - k);
    }
    prefixSum.set(currSum, (prefixSum.get(currSum) || 0) + 1);
  }
  return count;
};

File: LC-Python-Solution/560. Subarray Sum Equals K.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        
        count = 0
        curSum = 0
        numDic = {0:1}

        for num in nums:
            curSum += num

            if curSum-k in numDic:
                count+=numDic[curSum-k]
            
            numDic[curSum] = numDic.get(curSum,0)+1
        
        return count

        